[WHIT-2354] [WHIT-2355] Changes for farming grants

lauraghiorghisor-tw opened the pull request:
## Changes

### Finder changes for farming grants

- updated description of finder
- update document noun to "action"
- disabled documents summaries
- removed the email subscription option
- updated the summary text to include link to the capital grants finder

### Facet changes for farming grants

- "Payment type" facet: deleted
- New "Grant type" facet: added, users can only select one option, either "Countryside Stewardship Higher Tier" or "Sustainable Farming Incentive".
Land types
  - changed to include "Wood pasture"
  - previous "Woodland" -> "Woodland and forest"
Areas of interest
- New option:
   "Access",
   "Biodiversity",
   "Buffer strips",
   "Historic and archaeological features",
   "Orchards"
   "Species recovery and management"
   "Wildlife"
- Deleted option:
   "Air quality",
   "Crops",
   "Equipment and machinery",
   "Forestry",
    "Hedgerows",
   "Historic environment",
   "Land features and archaeological features",
   "Nutrient, manure and slurry management"
   "Priority habitats"
   "Public access and education"
- Changed options:
   "Flood risk" -> "Flood prevention",
   "Boundaries" -> "Field boundaries"
   "Pollinators and wildlife" -> "Pollinators"
   "Precision farming" -> "Precision farming, equipment and machinery"
   "Soil" -> "Soil health"

Named the new facet `farming_grant_type` (with prefix) as it overlaps with another `grant_type` in search api.
Changes at the bottom of the schema are just alphabetical ordering.

## Testing 

Tested locally using the instructions in https://docs.publishing.service.gov.uk/repos/govuk-docker/how-tos/finder-setup.html.

<img width="1158" height="801" alt="Screenshot 2025-08-13 at 09 15 49" src="https://github.com/user-attachments/assets/1da2b376-4593-45fd-b0da-267a7d4fccc0" />

<img width="1158" height="801" alt="Screenshot 2025-08-13 at 09 16 49" src="https://github.com/user-attachments/assets/8670afc6-9410-43fb-b321-5e4e27369985" />

<img width="1158" height="801" alt="Screenshot 2025-08-13 at 09 18 32" src="https://github.com/user-attachments/assets/df69bf8b-175b-4ed2-967a-67d7d7eecb52" />

<img width="656" height="462" alt="Screenshot 2025-08-13 at 09 18 54" src="https://github.com/user-attachments/assets/5354b93c-1645-4b47-b61d-e14c531cc2d2" />

<img width="656" height="462" alt="Screenshot 2025-08-13 at 09 19 22" src="https://github.com/user-attachments/assets/afbf181b-63a6-44b6-8990-1e4d167f52d5" />

<img width="656" height="340" alt="Screenshot 2025-08-13 at 09 19 35" src="https://github.com/user-attachments/assets/878cc2eb-355e-416b-a7a8-221b806588d7" />


Few examples of existing documents where the fields have been amended/deleted:
| Before | After |
| --- | --- |
| <img width="765" height="489" alt="prod" src="https://github.com/user-attachments/assets/b75634f4-7615-4120-af7b-e0b8ff70c0d1" /> | <img width="765" height="489" alt="new" src="https://github.com/user-attachments/assets/6e307b53-bcf9-4359-8170-0377fe9672ae" /> |
| <img width="765" height="489" alt="prod2" src="https://github.com/user-attachments/assets/d89c18eb-b26d-4f7d-97bd-8c9d3dd45dc8" /> | <img width="765" height="489" alt="new2" src="https://github.com/user-attachments/assets/5c126be1-a31c-4738-b2cd-c06fec687689" /> |
| <img width="765" height="489" alt="prod3" src="https://github.com/user-attachments/assets/767c708c-423e-452d-8972-de8c0e54da30" /> | <img width="765" height="489" alt="new3" src="https://github.com/user-attachments/assets/aca1b219-4270-48cb-b655-89771bfaa928" /> |
| <img width="765" height="489" alt="prod4" src="https://github.com/user-attachments/assets/deceb196-6c89-463b-81a5-204262c724c1" /> | <img width="765" height="489" alt="new4" src="https://github.com/user-attachments/assets/b3c78a2d-83ba-4728-8d9c-ecfab9eaf412" /> |

The issue we fixed in the UI:
<img width="656" height="340" alt="Farming issue" src="https://github.com/user-attachments/assets/3bd5debf-4396-49de-a6b7-3d96287515f6" />


---

[Jira1](https://gov-uk.atlassian.net/jira/software/c/projects/WHIT/boards/401?selectedIssue=WHIT-2355)
[Jira2](https://gov-uk.atlassian.net/jira/software/c/projects/WHIT/boards/401?selectedIssue=WHIT-2354)

lauraghiorghisor-tw committed:
Facet changes for farming grants

- "Payment type" facet: deleted
- New "Grant type" facet: added, users can only select one option, either "Countryside Stewardship Higher Tier" or "Sustainable Farming Incentive".
Land types
  - changed to include "Wood pasture"
  - previous "Woodland" -> "Woodland and forest"
Areas of interest
- New option:
   "Access",
   "Biodiversity",
   "Buffer strips",
   "Historic and archaeological features",
   "Orchards"
   "Species recovery and management"
   "Wildlife"
- Deleted option:
   "Air quality",
   "Crops",
   "Equipment and machinery",
   "Forestry",
    "Hedgerows",
   "Historic environment",
   "Land features and archaeological features",
   "Nutrient, manure and slurry management"
   "Priority habitats"
   "Public access and education"
- Changed options:
   "Flood risk" -> "Flood prevention",
   "Boundaries" -> "Field boundaries"
   "Pollinators and wildlife" -> "Pollinators"
   "Precision farming" -> "Precision farming, equipment and machinery"
   "Soil" -> "Soil health"

Named the new facet `farming_grant_type` (with prefix) as it overlaps with another `grant_type` in search api.
Changes at the bottom of the schema are just alphabetical ordering.

lauraghiorghisor-tw committed:
Finder changes for farming grants

- updated description of finder
- update document noun to "action"
- disabled documents summaries
- removed the email subscription option
- updated the summary text to include link to the capital grants finder

ryanb-gds commented on lib/documents/schemas/farming_grants.json:
```diff
@@ -2,10 +2,31 @@
-  "description": "Find out about grants and funding for farmers and land managers in England.",
-  "document_noun": "result",
+  "description": "Find out about grants and funding for farmers, foresters and land managers in England.",
+  "document_noun": "action",
```
Is this safe to change? I can't remember what this gets used for

lauraghiorghisor-tw replied:
This is used for the search results 🆗

ryanb-gds commented on lib/documents/schemas/farming_grants.json:
```diff
@@ -62,40 +87,32 @@
-          "label": "Historic environment",
-          "value": "historic-environment"
+          "label": "Historic and archaeological features",
+          "value": "historic-and-archaeological-features"
```
Should this have changed from historic-environment?

lauraghiorghisor-tw replied:
User confirmed this is ok.

ryanb-gds reviewed (approved):
Couple of things to check, but I imagine this will be alright

lauraghiorghisor-tw committed:
Do not show metadata for facets that are not allowed in the schema

Migrating the farming grants to a new schema left the old documents displaying odd metadata (it was rendering the entire array of values selected by the user). This ensures that, if the facet is not listed in the schema, we do not make an attempt to render its label in the summary.

We chose to omit the value altogether, rather than falling back to the old key.
